### PR TITLE
fix surface bug in all cases but in SP surface

### PR DESCRIPTION
**Pull request by yairchn**
This PR addresses issue 300
https://github.com/CliMA/TurbulenceConvection.jl/issues/300
In some cases a mixture of lowest face value (surface value) and lowest center value are mixed in the same Thermodynamic state.

### CODE REVIEW

File: src/Surface.jl [hunk 1]
```diff
@@ -112,6 +112,7 @@ function update(self::SurfaceBase{SurfaceFixedCoeffs}, GMV::GridMeanVariables)
     p0_f_surf = self.ref_state.p0[kf_surf]
     ρ0_f_surf = self.ref_state.rho0[kf_surf]
     α0_f_surf = self.ref_state.alpha0[kf_surf]
+    α0_c_surf = self.ref_state.alpha0_half[kf_surf]
```
Comment (charleskawczynski) [inline src/Surface.jl:115]:
```suggestion
    α0_c_surf = self.ref_state.alpha0_half[kc_surf]
```

File: src/Surface.jl [hunk 2]
```diff
@@ -156,6 +157,7 @@ function update(self::SurfaceBase{SurfaceMoninObukhov}, GMV::GridMeanVariables)
...
```
Comment (charleskawczynski) [inline src/Surface.jl:160]:
```suggestion
    α0_c_surf = self.ref_state.alpha0_half[kc_surf]
```

File: src/Surface.jl [hunk 3]
```diff
@@ -216,6 +218,7 @@ function update(self::SurfaceBase{SurfaceMoninObukhovDry}, GMV::GridMeanVariable
     p0_c_surf = self.ref_state.p0_half[kf_surf]
     ρ0_f_surf = self.ref_state.rho0[kf_surf]
     α0_f_surf = self.ref_state.alpha0[kf_surf]
+    α0_c_surf = self.ref_state.alpha0_half[kf_surf]
```
Comment (charleskawczynski) [inline src/Surface.jl:221]:
```suggestion
    α0_c_surf = self.ref_state.alpha0_half[kc_surf]
```

File: src/Surface.jl [hunk 4]
```diff
@@ -275,6 +278,7 @@ function update(self::SurfaceBase{SurfaceSullivanPatton}, GMV::GridMeanVariables
     p0_f_surf = self.ref_state.p0[kf_surf]
     p0_c_surf = self.ref_state.p0_half[kf_surf]
     ρ0_f_surf = self.ref_state.rho0[kf_surf]
+    ρ0_c_surf = self.ref_state.rho0_half[kf_surf]
```
Comment (charleskawczynski) [inline src/Surface.jl:281]:
```suggestion
    ρ0_c_surf = self.ref_state.rho0_half[kc_surf]
```

**Review by charleskawczynski — COMMENTED** [review]
Changes look good to me. Need to fix some of the indices. Also, can we delete the local variables `α0_f_surf`, I think?

**Comment by ilopezgp** [comment]
Can you write a description of why this change is needed?

**Comment by charleskawczynski** [comment]
Also, does this close #300?

File: src/Surface.jl [hunk 5]
```diff
@@ -131,7 +132,7 @@ function update(self::SurfaceBase{SurfaceFixedCoeffs}, GMV::GridMeanVariables)
     self.rho_hflux = -self.ch * windspeed * (θ_liq_ice_gm_surf - self.Tsurface / Π) * ρ0_f_surf
     self.shf = cp_ * self.rho_hflux
 
-    self.bflux = buoyancy_flux(param_set, self.shf, self.lhf, T_gm_surf, q_tot_gm_surf, α0_f_surf, ts)
+    self.bflux = buoyancy_flux(param_set, self.shf, self.lhf, T_gm_surf, q_tot_gm_surf, α0_c_surf, ts)
```
[the remark this answers is not in the document]
Comment (yairchn) [reply src/Surface.jl:135]: to the best of my understanding in 122 its at the surface 'ts = TD.PhaseEquil_pθq(param_set, p0_f_surf, self.Tsurface, self.qsurface)' 
and in 130 its at the center point 'ts = TD.PhaseEquil_pθq(param_set, p0_f_surf, θ_liq_ice_gm_surf, q_tot_gm_surf)'. The differences between near surface and surface values seem to be defining the flux:
'self.rho_hflux = -self.ch * windspeed * (θ_liq_ice_gm_surf - self.Tsurface / Π) * ρ0_f_surf'

[the remark this answers is not in the document]
Comment (yairchn) [reply src/Surface.jl:135]: But in more general terms this is what was brought in from SCAMPy. maybe we can leave these issues to the day we handle surface.

**Pull request closed without merging**